File: Data Structures & Algorithms/binary-tree-right-side-view/submission-6.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public List<int> RightSideView(TreeNode root) {
        var list = new List<int>();

        //add to queue
        //bfs (left -> right)
        //last remaining item should be right side
        //dont just take right
        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);

        while(queue.Any()){
            TreeNode right = null;
            var len = queue.Count;

            for(int i = 0; i < len; i++){
                var node = queue.Dequeue();

                //add left then right
                //handle null
                if(node != null){
                    right = node;
                    queue.Enqueue(right.left);
                    queue.Enqueue(right.right);
                }
                
            }

            if(right != null) list.Add(right.val);
        }

        return list;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-6.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        #edge case
        if root is None:
            return []
        

        queue = []
        res = []

        #add first item to queue
        queue.append(root)
        while queue:
            length = len(queue)
            currList = []
            for i in range(length):
                node = queue.pop(0)
                #check if node is not null
                currList.append(node.val)

                #add new item to end of queue
                if node.left != None:
                    queue.append(node.left)
                
                if node.right != None: 
                    queue.append(node.right)

            res.append(currList)
        
        return res
        
    

    #queue for level order
    #FIFO - Pop from front ( Pop(0) )